hack/generator/spec/crossplane.md

```diff
@@ -0,0 +1,71 @@
+# Thoughts on performing provider-azure code generation for Crossplane
+
+The patterns identified below were extrapolated from [the Crossplane VNET spec](https://github.com/crossplane/provider-azure/blob/master/config/crd/network.azure.crossplane.io_virtualnetworks.yaml).
```

Comment:
I'm afraid VNets might not be the best representation of our current CR schema patterns, which are best represented by v1beta1 resources. That said, I _think_ the key difference is probably the use of `spec.forProvider` and `spec.atProvider` as containers for the objects that map to the Azure API in v1beta1 resources.

https://github.com/crossplane/provider-azure/blob/master/config/crd/cache.azure.crossplane.io_redis.yaml